File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SquareOfSums(n int) int {
	var t int
	for x := 1; x <= n; x++ {
		t += x
	}
	return t*t
}

func SumOfSquares(n int) int {
	var t int
	for x := 1; x <= n; x++ {
		t += x*x
	}
	return t
}

func Difference(a int) int {
	return SquareOfSums(a) - SumOfSquares(a)
}

File: python/difference-of-squares/difference_of_squares.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

def square_of_sum(num):
    tot = 0
    for n in range(1, num + 1):
        tot += n
    return tot*tot

def sum_of_squares(num):
    tot = 0
    for n in range(1, num + 1):
        tot += n*n
    return tot

def difference(num):
    return square_of_sum(num) - sum_of_squares(num)
